app/components/tag/tag_component.rb

```diff
@@ -0,0 +1,12 @@
+class Tag::TagComponent < ViewComponent::Base
+  attr_reader :text
+
+  def initialize(text:, colour: nil)
+    @text = text
+    @colour = colour
+  end
+
+  def css_classes
+    @colour ? "govuk-tag--#{@colour}" : ''
```

Comment:
Should we validate the colours?

Reply:
Or should we just allow passing in `classes` like the nunjucks macro?

Reply:
The angle I took with the form builder was to provide arguments for the documented features of a component. [Here's a relevant example](https://github.com/DFE-Digital/govuk_design_system_formbuilder/blob/master/lib/govuk_design_system_formbuilder/traits/input.rb#L50) for configuring input width; in this case as a user, I'd probably want to be able to call something like `Tag::TagComponent.new(text: 'Danger', colour: 'red')`.

Reply:
There are mixed feelings on this. @adamsilver once tried doing something custom with a component in the Form builder that wasn't possible because you couldn't pass in custom `classes`. I ended up submitting a patch upstream: https://github.com/DFE-Digital/govuk_design_system_formbuilder/pull/96

The design system does allow you to pass custom classes for most of its components, and I think it's to allow users to break encapsulation where it's necessary, such as defining new variants of a component that are not yet upstream.

I think it makes sense to allow not only passing in `classes` but also providing smart arguments that interpret the specification in order to make components easy to use. So though it's more hard work, I think all 5 of these have their use-cases, and should be supported:

```ruby
GovukComponent::Tag.new(text: 'Danger', colour: 'red')

GovukComponent::Tag.new(text: 'Danger', colour: 'danger')

GovukComponent::Tag.new(text: 'Danger', classes: 'govuk-tag--red')
```

Reply:
If anything, the `'danger'` semantic variant could be dropped, I think it's the least important.

Reply:
What makes the nunjuck macros great is that they also allow you to pass in additional html attributes and the fixability it brings. [Heres some standards they use to determine the API for components](https://github.com/alphagov/govuk-frontend/blob/master/docs/contributing/coding-standards/nunjucks-api.md). I think we should inherit this to help use keep this gem up to date when the govuk-frontend makes breaking changes.

Reply:
The Nunjucks macros are powerful and flexible but I'm really not a fan of passing in strings of HTML. Invariably, from experience, you just end up with HTML in a string in a block of JavaScript/Ruby in a HTML(-like) template.